### PR TITLE
fix(cluster-credentials): behavior switch between providers

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

- Fixe to fetch credentials after changing cloud providers
- Clean code

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by evoxmusic** [comment]
Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://z2b87dda3-z1de3b556-gtw.za8ad0659.bool.sh)
:point_right: [storybook](https://z2b87dda3-z97e35ce4-gtw.za8ad0659.bool.sh)

File: libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx [hunk 1]
```diff
@@ -47,7 +42,7 @@ export function StepGeneral(props: StepGeneralProps) {
     [cloudProviders]
   )
 
-  const buildRegions: () => Value[] = useCallback(
```
Comment (bdebon) [inline libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx:50]: why did you change from useCallback to useMemo ? (it's for my understanding of the differences)

**Review by bdebon — APPROVED** [review]
works pretty well, just have a small question for my understanding

File: libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx:50]: Here, useMemo it's the better choice because useCallback returns a memoized function and useMemo returns a memoized value
